homeassistant/components/tahoma/__init__.py

```diff
@@ -111,6 +114,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
             hass.config_entries.async_forward_entry_setup(entry, platform)
         )
 
+    device_registry = await dr.async_get_registry(hass)
+
+    for gateway in gateways:
+        _LOGGER.debug(
+            "Added gateway (%s - %s - %s)",
+            gateway.id,
+            gateway.type,
+            gateway.sub_type,
+        )
+
+        gateway_model = (
+            beautify_name(gateway.sub_type.name)
+            if isinstance(gateway.sub_type, Enum)
+            else None
+        )
+
+        gateway_name = (
+            f"{beautify_name(gateway.type.name)} hub"
+            if isinstance(gateway.type, Enum)
+            else gateway.type
+        )
+
+        device_registry.async_get_or_create(
+            config_entry_id=entry.entry_id,
+            identifiers={(DOMAIN, gateway.id)},
```

Comment:
Do we know the MAC address of the device?  If so we can use connections instead

Reply:
Unfortunately not.. The Somfy account only communicates their internal device ids. We are already happy that the hostname of the device is using `gateway-xxxx-xxxx-xxx` where xxxx is the gateway id.

In theory we could gather the hostname via the discover possibilities (zeroconf / DHCP), but I am not sure if this works for all supported hubs.

Reply:
I will implement the gateway via zeroconf in a follow up PR. Resolving this comment for now.

Reply:
DHCP discovery added and tested in c70bfd4.